src/flushMicroTasks.ts

```diff
@@ -1,13 +1,5 @@
 import { setImmediate } from './helpers/timers';
 
-type Thenable<T> = { then: (callback: () => T) => unknown };
-
-export function flushMicroTasks(): Thenable<void> {
-  return {
-    // using "thenable" instead of a Promise, because otherwise it breaks when
-    // using "modern" fake timers
-    then(resolve) {
-      setImmediate(resolve);
-    },
-  };
+export function flushMicroTasks() {
+  return new Promise((resolve) => setImmediate(resolve));
```

Comment:
Yes, this is the most straightforward and correct way to wait for `setImmediate`. But the original comment said that this implementation breaks with modern fake timers? That turns out to be not true, doesn't it?

By the way, there's one more `setImmediate` await that could be replaced with `flushMicroTasks`, in the `waitForInternal` function.

Reply:
Can you test it with older versions of Jest and see where they fixed the implementation?